hs_app_timeseries/page_processors.py

```diff
@@ -1,106 +1,221 @@
-                             element_id=content_model.metadata.site.id if content_model.metadata.site else None)
 
-        variable_form = VariableForm(instance=content_model.metadata.variable, res_short_id=content_model.short_id,
-                             element_id=content_model.metadata.variable.id if content_model.metadata.variable else None)
+def _get_resource_edit_context(page, request, content_model, selected_series_id, series_ids,
+                               extended_metadata_exists):
+    site_form = None
+    variable_form = None
+    processing_level_form = None
+    method_form = None
+    timeseries_result_form = None
 
-        method_form = MethodForm(instance=content_model.metadata.method, res_short_id=content_model.short_id,
-                                 element_id=content_model.metadata.method.id if content_model.metadata.method else None)
+    if extended_metadata_exists:
+        # create timeseries sepecific metadata element forms
+        site = content_model.metadata.sites.filter(
+            series_ids__contains=[selected_series_id]).first()
+        site_form = SiteForm(instance=site, res_short_id=content_model.short_id,
+                             element_id=site.id if site else None,
+                             cv_site_types=content_model.metadata.cv_site_types.all(),
+                             cv_elevation_datums=content_model.metadata.cv_elevation_datums.all())
 
-        processing_level_form = ProcessingLevelForm(instance=content_model.metadata.processing_level,
+        site_form.action = _get_element_update_form_action('site', content_model.short_id, site.id)
+        site_form.number = site.id
+        site_form.set_dropdown_widgets(site_form.initial['site_type'],
+                                       site_form.initial['elevation_datum'])
+
+        variable = content_model.metadata.variables.filter(
+            series_ids__contains=[selected_series_id]).first()
+        variable_form = VariableForm(
+            instance=variable, res_short_id=content_model.short_id,
+            element_id=variable.id if variable else None,
+            cv_variable_types=content_model.metadata.cv_variable_types.all(),
+            cv_variable_names=content_model.metadata.cv_variable_names.all(),
+            cv_speciations=content_model.metadata.cv_speciations.all())
+
+        variable_form.action = _get_element_update_form_action('variable',
+                                                               content_model.short_id, site.id)
+        variable_form.number = variable.id
+        variable_form.set_dropdown_widgets(variable_form.initial['variable_type'],
+                                           variable_form.initial['variable_name'],
+                                           variable_form.initial['speciation'])
+
+        method = content_model.metadata.methods.filter(
+            series_ids__contains=[selected_series_id]).first()
```

Comment:
@pkdash Never mind, ignore my comments about missing pairing right parenthesis. I see it now.